core/src/components/progress-bar/progress-bar.ios.scss

```diff
@@ -1,8 +1,22 @@
 @import "./progress-bar";
+@import "./progress-bar.ios.vars";
 
 // iOS Progress bar
 // --------------------------------------------------
 
 :host {
-  height: 3px;
+  @include border-radius($progress-bar-ios-border-radius);
```

Comment:
In Figma and Xcode, this border radius applies both to the track (container) and the progress (value) bars. I think I remember some discussion about how we've decided not to round the corner of the progress bar because of animation challenges. Could we add that information to the design doc?

This PR:
![Screenshot 2023-12-27 at 10 00 55 AM](https://github.com/ionic-team/ionic-framework/assets/14926794/847b1154-511f-4fda-861f-4bef4abbda3b)


Xcode:
![Screenshot 2023-12-27 at 10 17 31 AM](https://github.com/ionic-team/ionic-framework/assets/14926794/abf4366d-7f49-497d-952e-1335185a032f)

Reply:
I can definitely add that extra context in the design document 👍 I put a note in the other info section of the PR for a brief amount of context:
> Accomplishing rounded caps for the inner progress with Ionic's scale transformation was not viable. To maintain 60fps animations for the progress, rounded corners were applied to the container.

I tried clip-path, pseudo elements, SVG cap elements for the edges of the progress, etc. All get skewed by the scale transformation on the actual progress bar. We are looking at a much larger overhaul to change the implementation from a scale transformation to something else (likely a translate would suffice).